### PR TITLE
add QD (C++/Fortran quad-double)

### PR DESCRIPTION
**Pull request by grlee77**
a C++/Fortran-90 double-double and quad-double package

This is an optional dependency of [Elemental](https://github.com/elemental/Elemental) (not yet packaged on conda-forge)

### CODE REVIEW

File: recipes/qd/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  sha256: c58e276f6fcf5f2f442c525f3de42ea00004734572b29c74028bbda0ad81096d
+
+build:
+  skip: true  # [win]
+  number: 0
+
+requirements:
+  build:
+    - toolchain
+    - gcc
```
Comment (isuruf) [inline recipes/qd/meta.yaml]: change this to `- gcc   # [osx]`

File: recipes/qd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  run:
+    - libgcc
```
Comment (isuruf) [inline recipes/qd/meta.yaml]: you only need `libgfortran` right?

Comment (grlee77) [inline recipes/qd/meta.yaml]: It is apparently also linking to libquadmath which I believe is included with libgcc?

Comment (isuruf) [inline recipes/qd/meta.yaml]: Ah. Didn't know that. @conda-forge/core can somebody review this?

Comment (jakirkham) [inline recipes/qd/meta.yaml]: Yeah for `libquadmath` one needs `libgcc`. Also we need to use `gcc` on all platforms then.

**Comment by isuruf** [comment]
Closing and reopening to trigger travis-ci

File: recipes/qd/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+test:
+  # make check in build.sh runs tests, these check the installation path
+  commands:
+    - test -d ${PREFIX}/include/qd
+    - test -f ${PREFIX}/lib/libqd.a
+    - test -f ${PREFIX}/lib/libqdmod.a
+    - test -f ${PREFIX}/lib/libqd_f_main.a
+    - test -f ${PREFIX}/bin/qd-config
```
Comment (jakirkham) [inline recipes/qd/meta.yaml]: Any reason not to run this program?

File: recipes/qd/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  run:
+    - libgcc
+
+test:
+  # make check in build.sh runs tests, these check the installation path
+  commands:
+    - test -d ${PREFIX}/include/qd
+    - test -f ${PREFIX}/lib/libqd.a
+    - test -f ${PREFIX}/lib/libqdmod.a
+    - test -f ${PREFIX}/lib/libqd_f_main.a
```
Comment (jakirkham) [inline recipes/qd/meta.yaml]: Are there `so`s and `dylib`s that we can check for?

File: recipes/qd/meta.yaml (hunk 2 above)
Comment (isuruf) [inline recipes/qd/meta.yaml]: How does that work? `gcc` is 4.8.3 and `libgcc` is 5.2.0 in the defaults channel.

Comment (jakirkham) [inline recipes/qd/meta.yaml]: 4.8.5 I think, but yes they are different.

IIUC `libgcc` 5 is backwards compatible with `libgcc` 4. The biggest changes for `gcc` were for C++11 support, but they provided both the old and new symbols in the new `libgcc`.

So not something we should need to worry about.

Comment (jakirkham) [inline recipes/qd/meta.yaml]: ref: https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html

File: recipes/qd/meta.yaml (hunk 1 above)
Comment (isuruf) [inline recipes/qd/meta.yaml]: ping

File: recipes/qd/build.sh [hunk 5]
```diff
@@ -1,10 +1,11 @@
 #!/bin/bash
 
+config_args="--enable-shared"
 if [[ `uname` == 'Darwin' ]]; then
     # make check below fails on osx unless $PREFIX/lib is added to rpath
-    LDFLAGS="$LDFLAGS -Wl,-rpath,${PREFIX}/lib" ./configure
+    LDFLAGS="$LDFLAGS -Wl,-rpath,${PREFIX}/lib" ./configure ${config_args}
```
Comment (isuruf) [inline recipes/qd/build.sh]: I think this is needed for Linux as well.

**Comment by isuruf** [comment]
@grlee77, I've pushed some changes. Hope you don't mind.

**Comment by grlee77** [comment]
LGTM. Thanks @isuruf!

**Pull request merged**